scripts/tests/test_validate_unwanted_patterns.py

```diff
@@ -0,0 +1,423 @@
+                        "of the end of the previous string."
+                    ),
+                )
+            ],
+        ),
+        (
+            (
+                """
+msg = (
+    "foo"
+    f" bar"
+)
+"""
+            ),
+            [
+                (
+                    3,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                )
+            ],
+        ),
+        (
+            (
+                """
+msg = (
+    f"foo"
+    f" bar"
+)
+"""
+            ),
+            [
+                (
+                    3,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                )
+            ],
+        ),
+        (
+            (
+                """
+msg = (
+    "foo"
+    rf" bar"
+    " baz"
+)
+"""
+            ),
+            [
+                (
+                    3,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+                (
+                    4,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+msg = (
+    "foo"
+    " bar"
+    rf" baz"
+)
+"""
+            ),
+            [
+                (
+                    3,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+                (
+                    4,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+msg = (
+    "foo"
+    rf" bar"
+    rf" baz"
+)
+"""
+            ),
+            [
+                (
+                    3,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+                (
+                    4,
+                    (
+                        "String has a space at the beginning instead "
+                        "of the end of the previous string."
+                    ),
+                ),
+            ],
+        ),
+    ],
+)
+def test_strings_with_wrong_placed_whitespace(data, expected):
```

Comment:
Same comment; if you can split into acceptable patterns versus failing patterns will make much more readable